openmc/kinetics/aux_functions.py

```diff
@@ -0,0 +1,218 @@
+    vs = view.shape
+    view.shape = (to_shape[0], ratios[0], vs[1], to_shape[2], ratios[2],vs[3],vs[4])
+    to_array[...,4:6] = np.sum(view, axis=(1,4))
+
+    # Sum y-max currents
+    view = array[:,ratios[1]-1:from_shape[1]:ratios[1],:,:,6:8]
+    vs = view.shape
+    view.shape = (to_shape[0], ratios[0], vs[1], to_shape[2], ratios[2],vs[3],vs[4])
+    to_array[...,6:8] = np.sum(view, axis=(1,4))
+
+    # Sum z-min currents
+    view = array[0:from_shape[0]:ratios[0],:,:,:,8:10]
+    vs = view.shape
+    view.shape = (vs[0], to_shape[1], ratios[1], to_shape[2], ratios[2],vs[3],vs[4])
+    to_array[...,8:10] = np.sum(view, axis=(2,4))
+
+    # Sum z-max currents
+    view = array[ratios[0]-1:from_shape[0]:ratios[0],:,:,:,10:12]
+    vs = view.shape
+    view.shape = (vs[0], to_shape[1], ratios[1], to_shape[2], ratios[2],vs[3],vs[4])
+    to_array[...,10:12] = np.sum(view, axis=(2,4))
+
+    to_array.shape = to_shape
+    return to_array
+
+
+def block_view(A, block):
+    """Provide a block view to 3+D array. No error checking made.
+    Therefore meaningful (as implemented) only for blocks strictly
+    compatible with the shape of A."""
+
+    block_dim = len(block)
+
+    # Check that blocks fit nicely into first dimensions of A
+    if block_dim > len(A.shape):
+        msg = 'Cannot access block_view of {}-D array with block size {}-D'\
+              ' that is greater than array dimensions'.format(len(A.shape),
+                                                              block_dim)
+        raise ValueError(msg)
+    else:
+        block = block + A.shape[block_dim:]
+        block_dim = len(A.shape)
+
+    for i,b in enumerate(block):
+        if A.shape[i] % b != 0:
+            msg = 'Cannot access block_view of {}-D array with block {}'\
+                  ' since block dimension {} does not fit evenly in'\
+                  ' Array dimension {}'.format(len(A.shape), block,
+                                               b, A.shape[i])
+            raise ValueError(msg)
+
+    # simple shape and strides computations may seem at first strange
+    # unless one is able to recognize the 'tuple additions' involved
+    shape = ()
+    strides = ()
+
+    for i,b in enumerate(block):
+        shape = shape + (int(A.shape[i] / block[i]),)
+        strides = strides + (b * A.strides[i],)
+
+    shape = shape + block
+    strides = strides + A.strides
+
+    return ast(A, shape=shape, strides=strides)
+
+def nan_inf_to_zero(array):
+
+    array[array == -np.inf] = 0.
+    array[array ==  np.inf] = 0.
+    return np.nan_to_num(array)
+
+def compute_eigenvalue(A, M, flux, tolerance=1.e-6):
+
+    # Ensure flux is a 1D array
+    flux = flux.flatten()
+
+    # Compute the initial source
+    old_source = M * flux
+    norm = old_source.mean()
+    old_source  = old_source / norm
+    flux  = flux / norm
+    k_eff = 1.0
+
+    print('initial k {}'.format((M*flux).sum() / (A*flux).sum()))
+
+    for i in range(10000):
```

Comment:
Let's name that constant somewhere, like `max_eigenvalue_iterations` or something. In fact, it should probably be a kwarg to this function with that default value.